src/Button/index.js

```diff
@@ -28,21 +30,94 @@ class Button extends React.PureComponent {
 	};
 
 	static defaultProps = {
-		disabled: false
+		disabled: false,
+		onHoldCallStart: 300,
+		onHoldCallStep: 50,
 	};
 
 	constructor(props) {
 		super(props);
+		this.node = React.createRef();
+
 		this.state = {
 			focused: false,
-			menuOpen: false
+			menuOpen: false,
+			held: false
 		};
 
 		this.onBlur = this.onBlur.bind(this);
 		this.onClick = this.onClick.bind(this);
 		this.onKeyPress = this.onKeyPress.bind(this);
+
+		this.onMouseOut = this.onMouseOut.bind(this);
+		this.onMouseDown = this.onMouseDown.bind(this);
+		this.onTouchStart = this.onTouchStart.bind(this);
+		this.onTouchEnd = this.onTouchEnd.bind(this);
+	}
+
+	componentDidMount() {
+		this.node.current.addEventListener('touchstart', this.onTouchStart);
+		this.node.current.addEventListener('touchend', this.onTouchEnd);
+	}
+
+	componentWillUnmount() {
+		this.node.current.removeEventListener('touchstart', this.onTouchStart);
+		this.node.current.removeEventListener('touchend', this.onTouchEnd);
 	}
 
+	componentDidUpdate(prevProps, prevState, snapshot) {
+		if (this.props.disabled && this.state.held) {
+			this.clearHoldTimeout();
+		}
+	}
+
+	onTouchStart(evt) {
+		evt.stopPropagation();
```

Comment:
Is it ok to stop propagation and/or preventDefault, when we're not actually doing anything ourselves?
E.g. if there's no onHold handler, don't we want long click to be a click?

Reply:
(maybe one or both of those should be inside the if, or maybe if we add listeners when we have a handler, it'll be a moot point)